bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs/diff/MergeNode.scala

```diff
@@ -113,11 +120,105 @@ object MergeNode {
+
+    def focus(k: SPNodeKey): TryVcs[TreeLoc[MergeNode]] = t.loc.findNode(k)
+
+    def mRootLabel: TryVcs[Modified] =
+      t.rootLabel match {
+        case m: Modified => m.right
+        case l           => TryVcs.fail(s"Expected Modified label for ${l.key}")
+      }
+
+    def mModifyLabel(f: Modified => Modified): TryVcs[Tree[MergeNode]] =
+      t.rootLabel match {
+        case m: Modified => Tree.node(f(m): MergeNode, t.subForest).right
+        case _           => TryVcs.fail(s"Expected Modified label for $key")
+      }
   }
 
+  implicit class MergeZipperOps(z: TreeLoc[MergeNode]) {
+    def key: SPNodeKey = z.getLabel.key
+
+    def findNode(k: SPNodeKey): TryVcs[TreeLoc[MergeNode]] =
+      z.find(_.key === k).toTryVcs(s"Could not find descendant $k")
+
+    def incr(lifespanId: LifespanId): TryVcs[TreeLoc[MergeNode]] =
+      mModifyLabel(m => m.copy(nv = m.nv.incr(lifespanId)))
+
+    // Determine the index of the focused node in its parent, if any.
+    def childIndex: TryVcs[Int] =
+      z.parent.map { _.tree.subForest.indexWhere(_.key === key) }.filter(_ >= 0).toTryVcs(s"Could not find child index of $key")
+
+    // Ensure that the node at the focus is a `Modified` node, converting an
+    // `Unmodified` node to `Modified` if necessary.  It should be the case that
+    // all `Unmodified` nodes appear in the `nodeMap`.  If not, the conversion
+    // will fail with a left.
+    def asModified(nodeMap: Map[SPNodeKey, ISPNode]): TryVcs[TreeLoc[MergeNode]] =
+      mapAsModified(nodeMap)(identity)
+
+    // Update the given node, converting it to a Modified node if necessary.
+    // Sorry, this is a bit awkward.
+    def mapAsModified(nodeMap: Map[SPNodeKey, ISPNode])(f: Modified => Modified): TryVcs[TreeLoc[MergeNode]] =
+      z.getLabel match {
+        case m: Modified   =>
+          TryVcs(z.modifyLabel(_ => f(m)))
+
+        case Unmodified(k) =>
+          nodeMap.get(k).toTryVcs(s"Unmodified node $k not found in node map.").map { n =>
+            z.modifyTree { _ =>
+              val mod = f(Modified(n.key, n.getVersion, n.getDataObject, NodeDetail(n)))
+              Tree.node(mod, n.children.map(c => unmodified(c).leaf).toStream)
+            }
+          }
+      }
+
+    def mModifyLabel(f: Modified => Modified): TryVcs[TreeLoc[MergeNode]] =
+      z.tree.mModifyLabel(f).map { t => z.modifyTree(_ => t) }
+
+    def getOrCreateConflictFolder(lifespanId: LifespanId, nodeMap: Map[SPNodeKey, ISPNode]): TryVcs[TreeLoc[MergeNode]] = {
+      def isConflictFolder(t: Tree[MergeNode]): Boolean =
+        t.rootLabel match {
+          case Modified(_,_, _: ConflictFolder, _) => true
+          case Unmodified(k)                       => nodeMap.get(k).exists { n =>
+            n.getDataObject.getType == SPComponentType.CONFLICT_FOLDER
+          }
+          case _                                   => false
+        }
+
+      //noinspection MutatorLikeMethodIsParameterless
```

Comment:
Yeah, it keeps Idea from warning.  It may have a mutator like name but it doesn't mutate anything.  It produces a new value.